src/main/java/org/opentripplanner/middleware/utils/ItineraryExistenceChecker.java

```diff
@@ -0,0 +1,66 @@
+package org.opentripplanner.middleware.utils;
+
+import org.opentripplanner.middleware.otp.OtpDispatcher;
+import org.opentripplanner.middleware.otp.OtpDispatcherResponse;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.OtpResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * This utility class checks for existence of itineraries for given OTP queries.
+ */
+public class ItineraryExistenceChecker {
+    private final Function<String, OtpDispatcherResponse> otpDispatcherFunction;
+
+    /**
+     * @param otpDispatcherFunction a function that takes String as input and returns {@link OtpDispatcherResponse},
+     *                              example: {@link OtpDispatcher#sendOtpPlanRequest}.
+     */
+    public ItineraryExistenceChecker(Function<String, OtpDispatcherResponse> otpDispatcherFunction) {
+        if (otpDispatcherFunction == null) throw new NullPointerException();
+        this.otpDispatcherFunction = otpDispatcherFunction;
+    }
+
+    /**
+     * Checks that, for each query provided, an itinerary exists.
+     * @param labeledQueries a map containing the queries to check, each query having a key or
+     *                       label that will be used in Result.labeledResponses for easy identification.
+     * @return An object with a map of results and summary of itinerary existence.
+     */
+    public Result checkAll(Map<String, String> labeledQueries, boolean checkArrival) {
+        // TODO: Consider multi-threading?
+        Map<String, OtpResponse> responses = new HashMap<>();
+        boolean allItinerariesExist = true;
+
+        for (Map.Entry<String, String> entry : labeledQueries.entrySet()) {
+            OtpDispatcherResponse response = otpDispatcherFunction.apply(entry.getValue());
+            responses.put(entry.getKey(), response.getResponse());
+
+            Itinerary sameDayItinerary = response.findItineraryDepartingSameDay(checkArrival);
+            if (sameDayItinerary == null) allItinerariesExist = false;
+        }
+
+        return new Result(allItinerariesExist, responses);
+    }
+
+    /**
+     * Class to pass the results of the OTP itinerary checks.
+     */
+    public static class Result {
+        /** Whether all itineraries checked exist. */
+        public final boolean allItinerariesExist;
+        /**
+         * A map with the same keys as the input from checkAll,
+         * and values as OTP responses for the corresponding queries.
+         */
+        public final Map<String, OtpResponse> labeledResponses;
+
+        private Result(boolean itinerariesExist, Map<String, OtpResponse> labeledResponses) {
+            this.allItinerariesExist = itinerariesExist;
+            this.labeledResponses = labeledResponses;
+        }
+    }
+}
```

Comment:
This file is small enough that it'd probably be fine to merge it into the `ItineraryUtils` class.

Reply:
Merged in https://github.com/ibi-group/otp-middleware/pull/77/commits/91812109313baeba91b81c2fea66fb1fa954cca9.